cpp/src/arrow/compute/kernel.h

```diff
@@ -386,47 +435,75 @@ struct SimdLevel {
   enum type { NONE, SSE4_2, AVX, AVX2, AVX512, NEON };
 };
 
+/// \brief The strategy to use for propagating or otherwise populating the
+/// validity bitmap of a kernel output.
 struct NullHandling {
   enum type {
     /// Compute the output validity bitmap by intersecting the validity bitmaps
-    /// of the arguments. Kernel does not do anything with the bitmap
+    /// of the arguments. Kernel generally need not touch the bitmap
+    /// thereafter, but a kernel's exec function is permitted to alter the
+    /// bitmap after the null intersection is computed if it needs to.
```

Comment:
From this explanation, I find it not fully clear what "intersecting" means. 
(is it where one of the arguments is null, or all or the argument is null, that the output validity indicates null?)

Reply:
Yes. I'll clarify that it means a bitwise-and operation with all of the respective validity bitmaps